pkg/debuginfo/store.go

```diff
@@ -97,6 +101,9 @@ func NewStore(
 		cacheDir:         cacheDir,
 		metadata:         metadata,
 		debuginfodClient: debuginfodClient,
+
+		existsCacheMtx: &sync.RWMutex{},
+		existsCache:    map[string]struct{}{},
```

Comment:
I'm worried about these caches being unbounded. While it's not very likely that the Agent will run for an extremely long time without restarts, could we set a limit on it with perhaps an LRU eviction policy?

Reply:
Mmm, let's maybe do 10k? Sorry if I spoke too soon:

Assuming a key size of 20 characters, that hashmaps have no overhead, not counting for the locking, and that strings take 8 bytes per char (not really true):

(20 characters * 8 bytes per char + 1 byte for the value )* 100k elements = ~16MB